provider/sidecar.go

```diff
@@ -92,11 +106,8 @@ func (provider *Sidecar) Provide(configurationChan chan<- types.ConfigMessage, p
 	if err != nil {
 		log.Fatalln("Error reloading Sidecar config", err)
 	}
-	err = provider.loadSidecarConfig(states.ByService())
-	if err != nil {
-		return err
-	}
-	return nil
+
+	return provider.loadSidecarConfig(states.ByService())
```

Comment:
As mentioned in #7, I'd just pass `states` here and loop over them with `states.EachService()` later. Saves allocation for a biggish map and lots less CPU time.